ENS/ens-beginners-guide.md

```diff
@@ -0,0 +1,222 @@
+
+### Prerequisites
+
+First you will need to install Node.js and npm by following the install instructions on the [Node.js website](https://nodejs.org/ "Node.js").
+
+If you are not running a full mainnet Ethereum node, you will need to setup an Infura endpoint for connecting. This is easy to do - just sign up on the [infura.io](https://infura.io) website. Create a new project and copy the mainnet URL. It will be something like this: `mainnet.infura.io/v3/<your project id>`.
+
+Now set your Infura URL including `https://` at the beginning, as an environment variable. This will only set it for the current session so if you close your terminal window you will need to do it again.
+
+On Linux or macOS:
+```shell
+export INFURA_URL=https://mainnet.infura.io/v3/<your project id>
+```
+
+On Windows:
+```bash
+set INFURA_URL=https://mainnet.infura.io/v3/<your project id>
+```
+
+Create a new project directory called `enslookup` and change directory into it.
+
+```bash
+mkdir enslookup
+cd enslookup
+```
+
+### Web3.js
+
+We will be using Web3.js to interact with the ENS registry. To install Web3.js use the Node.js package manager `npm`.
+
+Create a file called `package.json`:
+
+```json
+{
+    "dependencies": {
+        "web3": "1.0.0-beta.50"
+    }
+}
+```
+
+Then install the package by executing:
+
+```bash
+npm install
+```
+
+### Enslookup
+
+Create a new file called `enslookup` (notice there is no file extension).
+
+If you are on a Linux or macOS, make the file executable:
+```bash
+chmod +x enslookup
+```
+
+Edit the file in your favorite editor - don't worry too much about the code here but it:
+ - Allows the file to be executed by Node.js
+ - Allows us to use Javascript async/await syntax
+ - Gets the ENS name from the command's parameters
+ - Checks that you have set the Infura URL environment variable
+
+```js
+#!/usr/bin/env node
+
+(async () => {
+
+    // get the ens name to lookup as the first argument to the command
+    const [,,name] = process.argv;
+    // bail if no name provided
+    if(!name){
+        console.error('No name provided for lookup\nUsage: enslookup <name>')
+        return;
+    }
+    // bail if no infura url provided as environment variable
+    if(!process.env.INFURA_URL){
+        console.error(`Missing INFURA_URL environment variable`);
+        return;
+    }
+    console.log(`Name:\t\t${name}`);
+
+    // insert more cool stuff here
+
+})();
+```
+
+Next, we will setup Web3.js to connect to Ethereum:
+
+```js
+// load web3 library
+const Web3 = require('web3');
+
+// connect to an Infura endpoint to connect to Ethereum (passed as Environment Variable)
+// feel free to use your own mainnet node
+const web3 = new Web3(process.env.INFURA_URL);
+```
+
+Now, it is very tempting to jump right in and lookup an address using `web3.eth.ens.getAddress(EnsName)` but there is something important to consider first.
```

Comment:
Another small example of a "weasel word", try:

```suggestion
Now, it is tempting to jump right in and lookup an address using `web3.eth.ens.getAddress(EnsName)` but there is something important to consider first.
```

Reply:
Removed